libraries/bezier-rs/src/subpath/transform.rs

```diff
@@ -272,6 +274,183 @@ impl<ManipulatorGroupId: crate::Identifier> Subpath<ManipulatorGroupId> {
+				// the points as being on top of one another.
+				if let Some((clipped_subpath1, clipped_subpath2)) = Subpath::clip_subpaths(subpath1, subpath2) {
+					subpaths[i] = clipped_subpath1;
+					subpaths[j] = clipped_subpath2;
+					apply_joint = false;
+				}
+			}
+			// The angle is convex. The Subpath must be joined using the specified Joint type
+			if apply_joint {
+				match joint {
+					Joint::Mitre => todo!(),
+					Joint::Rounded => todo!(),
+					Joint::Blunt => {
+						drop_common_point[j] = false;
+					}
+				}
+			}
+		}
+
+		// Clip any overlap in the last segment
+		if self.closed {
+			let out_tangent = self.get_segment(self.len_segments() - 1).unwrap().tangent(TValue::Parametric(1.));
+			let in_tangent = self.get_segment(0).unwrap().tangent(TValue::Parametric(0.));
+			let angle = out_tangent.angle_between(in_tangent);
+
+			let mut apply_joint = true;
+			if (angle > 0. && distance > 0.) || (angle < 0. && distance < 0.) {
+				if let Some((clipped_subpath1, clipped_subpath2)) = Subpath::clip_subpaths(&subpaths[subpaths.len() - 1], &subpaths[0]) {
+					// Merge the clipped subpaths
+					let last_index = subpaths.len() - 1;
+					subpaths[last_index] = clipped_subpath1;
+					subpaths[0] = clipped_subpath2;
+					apply_joint = false;
+				}
+			}
+			if apply_joint {
+				match joint {
+					Joint::Mitre => todo!(),
+					Joint::Rounded => todo!(),
+					Joint::Blunt => {
+						drop_common_point[0] = false;
+					}
+				}
+			}
+		}
+
+		// Merge the subpaths. Drop points which overlap with one another.
+		let mut manipulator_groups = subpaths[0].manipulator_groups.clone();
+		for i in 1..subpaths.len() {
+			if drop_common_point[i] {
+				let last_group = manipulator_groups.pop().unwrap();
+				let mut manipulators_copy = subpaths[i].manipulator_groups.clone();
+				manipulators_copy[0].in_handle = last_group.in_handle;
+
+				manipulator_groups.append(&mut manipulators_copy);
+			} else {
+				manipulator_groups.append(&mut subpaths[i].manipulator_groups.clone());
+			}
+		}
+		if self.closed && drop_common_point[0] {
+			let last_group = manipulator_groups.pop().unwrap();
+			manipulator_groups[0].in_handle = last_group.in_handle;
+		}
+
+		Subpath::new(manipulator_groups, self.closed)
+	}
+
+	/// Outline returns a single closed subpath (if the original subpath was open) or two closed subpaths (if the original subpath was closed) that forms
+	/// an approximate outline around the subpath at a specified distance from the curve. Outline takes the following parameters:
+	/// - `distance` - The outline's distance from the curve.
+	/// - `joint` - The joint type used to cap the endpoints of open bezier curves, and join successive subpath segments.
+	/// <iframe frameBorder="0" width="100%" height="375px" src="https://graphite.rs/bezier-rs-demos#subpath/outline/solo" title="Outline Demo"></iframe>
+	pub fn outline(&self, distance: f64, joint: Joint) -> (Subpath<ManipulatorGroupId>, Option<Subpath<ManipulatorGroupId>>) {
```

Comment:
Ah, right, here's a case where we begin needing the `Path` data type (which is composed of multiple `Subpath`s) so a single result can be returned that might have a hole in it (for the case when the input Subpath is closed). But for now, just add a TODO comment to change this function's return type to a Path once we have that data type so it can represent the potential hole.

Reply:
Let's chat about the `Path` data type in our next meeting. Perhaps we could add something relatively simple in the short term (in a future PR of course).